### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by test0n3**
- Page admin created
- admin shows list more active users by last_week and by last_month
- CSV file can be created and downloaded upon request by link

### CODE REVIEW

File: controllers/generateCSV.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+def generateCSV
+  # name archive by date
+  filename = Time.now.strftime("%Y%m%d")
+
+  # get json and parse it
+  # my_hashes = JSON.parse(File.open("data.json").read)
+
+  # using class User to parse the Json file
+  user = User.all
```
Comment (condef5) [inline controllers/generateCSV.rb]: Your could rename this variable as `users`, in plural because method return array.

Comment (test0n3) [reply controllers/generateCSV.rb]: Change will be noticeable in the next push.

File: controllers/generateCSV.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  cell_title = ["email", "name", "genre", "set_milestone", "date", "height", "weight"]
+
+  # open CSV file
+  # CSV.open("./public/#{filename}.csv", "w") do |csv|
+  CSV.generate do |csv|
+    # write completed column names
+    csv << cell_title
+  # using main defined classes
+    user.each do |userdata|
+      userdata.measures.each do |each_measure|
```
Comment (condef5) [inline controllers/generateCSV.rb]: Your variable `each_measure` should be named `measure` because it's the singular form of the collection.

Comment (test0n3) [reply controllers/generateCSV.rb]: Chances will be noticeable in the next push.

File: routes.rb [hunk 3]
```diff
@@ -1,9 +1,11 @@
 require 'sinatra'
 require 'sinatra/reloader' # reload server
 require 'sinatra/flash'
-require './models/User'
+require_relative './models/User'
```
Comment (luciagirasoles) [inline routes.rb]: '.' is not necessary as it is relative

File: routes.rb [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
+require_relative "./controllers/generateCSV"
```
Comment (luciagirasoles) [inline routes.rb]: '.' is not necessary as it is relative

**Pull request merged**